### ISSUE TITLE
Test Methods With No Assert

### ISSUE THREAD

**Issue by ghost** [SFDO-Community/declarative-lookup-rollup-summaries#581]
The Apex Test Methods do not Assert that the code behaves properly. Verifying that your code works as you expect it to work is the most important part of unit testing. It is important to ensure that all test methods use the System.assert methods to prove that the code behaves as expected. If you do not use any System.assert() method calls, then the tests are not verifying results properly.
References:
https://wiki.developerforce.com/index.php/An_Introduction_to_Apex_Code_Test_Methods
https://wiki.developerforce.com/index.php/How_to_Write_Good_Unit_Tests

**Comment by DanDonin** [comment]
I just poked through a handful of the test classes and they do have asserts.  What test class are you specifically saying isn't using asserts?

**Comment by jschneider-nexgen** [comment]
dlrs_AccountTest (standard object)
```
`/**
 * Auto Generated and Deployed by the Declarative Lookup Rollup Summaries Tool package (dlrs)
 **/
@IsTest
private class dlrs_AccountTest
{
    @IsTest
    private static void testTrigger()
    {
        // Force the dlrs_AccountTrigger to be invoked, fails the test if org config or other Apex code prevents this.
        dlrs.RollupService.testHandler(new Account());
    }
}
```
dlrs_PolicyTest (custom object)
```
/**
 * Auto Generated and Deployed by the Declarative Lookup Rollup Summaries Tool package (dlrs)
 **/
@IsTest
private class dlrs_PolicyTest
{
    @IsTest
    private static void testTrigger()
    {
        // Force the dlrs_PolicyTrigger to be invoked, fails the test if org config or other Apex code prevents this.
        dlrs.RollupService.testHandler(new Policy__c());
    }
}
```

**Comment by DanDonin** [comment]
Yeah it passes the rollup into test classes built into the package where
all the asserts live. This allows the package to generate test dynamically
as there are infinite dlrs records/triggers that people could create.  If
you go into the test classes that are namespaced with dlrs you can see the
bulk of the test code that is run.

**Comment by afawcett** [comment]
Yes, @DanDonin is spot on, in this case the packaged tests assert the quality of the code, not this generated test. Adding asserts to the generated test code in this case would also be challenging, since there is no way to setup all the required data automatically without knowing the use case. At @jschneider-nexgen does this help as regards your original need? Is someone querying you on this? We might be able to add comments to the tests or a fake assert to keep code scanners happy?

**Comment by jschneider-nexgen** [comment]
The problem I run into with it is when a client runs the Force.com Source Scanner Report (http://lp.checkmarx.com/salesforce/) (https://security.secure.force.com/security/tools/forcecom/scannerhelp) and asks about these test classes.  Personally, I get it, but a fake assert would likely resolve this and prevent the conversation.

**Comment by afawcett** [comment]
Ah see, they must have install the code unmanaged then, rather than as a managed package? Otherwise the Source Scanner would not see the test code. Happy to flag this as an enhancement though.

**Comment by aheber** [comment]
This will be delivered in our new LWC trigger management(available in the upcoming v2.26). It will not be backported to the Visualforce interface.
It will certainly not be more meaningful than "didn't crash before now" but it should satisfy linters.
